sensor/debugger/k8s/fake.go

```diff
@@ -318,9 +400,48 @@ func (f *FakeEventsManager) createEvent(msg resources.InformerK8sMsg, ch chan<-
 }
 
 // waitOnMode waits depending on the mode
-func (f *FakeEventsManager) waitOnMode() {
+func (f *FakeEventsManager) waitOnMode(events []string) error {
 	switch f.Mode {
 	case Delay:
 		time.Sleep(f.Delay)
+		return nil
+	case ChannelAck:
+		if len(events) == 0 {
+			return nil
+		}
+		receivedEvents := 0
+		for {
+			timeout := time.After(5 * time.Second)
+			select {
+			case <-timeout:
+				return errors.New("timeout reached waiting for event")
+			case event, more := <-f.AckChannel:
+				if !more {
+					return errors.New("channel closed")
+				}
```

Comment:
Is it really an error? In which situations the channel gets closed prematurely?

Reply:
Found it - we close the channel when the tests are done. So this is not an error - I suggest to: (1) ignore this case, (2) return nil. I think, that option (2) maybe cleaner.

Reply:
I decided to ignore as you suggested. If we return nil we can land in a situation where the channel is closed and we keep processing events without waiting thus making the results of the test unpredictable.